File: archived/src/erlang/ch01/1.2/l1/src/exercises_starter.erl
-module(exercises_starter).
-export([exercise1/0, exercise2/0, exercise3/2, exercise4/1, exercise5/1, main/0]).

exercise1() -> {100 div 9, 100 rem 9}.
exercise2() -> {123 div 11, 123 rem 11}.
exercise3(N, D) ->
    Q = N div D, R = N rem D,
    N =:= D*Q + R.
exercise4(N) -> N band 15.  %% N % 16
exercise5(N) -> (N rem 7) =:= 0.

main() ->
    {Q1,R1} = exercise1(),
    io:format("Exercise 1 (100//9,100%9): ~p ~p~n", [Q1,R1]),
    {Q2,R2} = exercise2(),
    io:format("Exercise 2 (123//11,123%11): ~p ~p~n", [Q2,R2]),
    io:format("Exercise 3 check (200,23): ~p~n", [exercise3(200,23)]),
    io:format("Exercise 4 (37 % 16 via bitmask): ~p~n", [exercise4(37)]),
    io:format("Exercise 5 (35 divisible by 7?): ~p~n", [exercise5(35)]).

File: archived/src/go/ch01/1.2/l1/exercises_starter.go
package main

import "fmt"

func ex1() (int, int) { return 100 / 9, 100 % 9 }
func ex2() (int, int) { return 123 / 11, 123 % 11 }
func ex3(n, d int) bool {
	q, r := n/d, n%d
	return n == d*q+r
}
func ex4(n int) int { return n & 15 }        // n % 16
func ex5(n int) bool { return n%7 == 0 }     // divisibility by 7 (simple)

func runExercises() {
	q, r := ex1()
	fmt.Println("Exercise 1 (100//9,100%9):", q, r)
	q, r = ex2()
	fmt.Println("Exercise 2 (123//11,123%11):", q, r)
	fmt.Println("Exercise 3 check (200,23):", ex3(200, 23))
	fmt.Println("Exercise 4 (37 % 16 via bitmask):", ex4(37))
	fmt.Println("Exercise 5 (35 divisible by 7?):", ex5(35))
}
